synapse/storage/databases/main/deviceinbox.py

```diff
@@ -261,12 +273,19 @@ def get_new_messages_for_remote_destination_txn(txn):
             )
             txn.execute(sql, (destination, last_stream_id, current_stream_id, limit))
             messages = []
```

Comment:
Do we need to instantiate `stream_pos` like we do above?

It's a shame that this code looks identical minus the table name.

Reply:
Oh! I didn't notice that until now. Refactoring it is definitely an option.

Reply:
Yep, refactoring is more of a long term thing, just noticed it while reviewing. (I didn't closely inspect if they *are* identical by the way. They might just be very similar.)